### PR TITLE
support batch-size prediction

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
### PR types

Function optimization
### PR changes

APIs
### Describe

主要修改点：
1. 完善 `Solver.predict` 方法，支持按 batch size 进行预测

其他：
1. 修复 `ModelList.forward` 多次调用 transform的问题，改为直接调用 `submodel.forward`
2. 修复cylinder2d_Re100文档中的一处代码引用错误

### CODE REVIEW

File: ppsci/solver/solver.py [hunk 1]
```diff
@@ -420,9 +420,54 @@ def visualize(self, epoch_id=0):
...
+                    batch_output_dict = self.model(batch_input_dict)
+            else:
+                batch_output_dict = self.model(batch_input_dict)
+
+            # collect batch data
+            for key, batch_output in batch_output_dict.items():
+                pred_dict[key].append(batch_output)
+
+        pred_dict = {
+            key: paddle.concat(value, axis=-1) for key, value in pred_dict.items()
```
Comment (zhiminzhang0830) [inline ppsci/solver/solver.py:468]: 此处是否因该为：`paddle.concat(value, axis=0)`？

Comment (HydrogenSulfate) [reply ppsci/solver/solver.py]: 已修改

File: ppsci/solver/solver.py [hunk 2]
```diff
@@ -420,9 +420,52 @@ def visualize(self, epoch_id=0):
...
+    def predict(self, input_dict: Dict[str, paddle.Tensor], batch_size: int = 64):
+        """Pure prediction using model.forward(...), support single device prediction yet.
+
+        Args:
+            input_dict (Dict[str, paddle.Tensor]): Input data in dict.
+            batch_size (int, optional): Predicting by batch size. Defaults to 64.
+
+        Returns:
+            Dict[str, paddle.Tensor]: Prediction in dict.
+        """
```
Comment (zhiminzhang0830) [inline ppsci/solver/solver.py:433]: 此函数内需要将model转成eval模式 `self.model.eval()`?

**Review by zhiminzhang0830 — APPROVED** [review]
LGTM
